app/models/course/condition/achievement.rb

```diff
@@ -3,4 +3,46 @@ class Course::Condition::Achievement < ActiveRecord::Base
   belongs_to :achievement, class_name: Course::Achievement.name, inverse_of: false
 
   default_scope { includes(:achievement) }
+
+  validate :validate_achievement_condition, if: :achievement_id_changed?
```

Comment:
(Wrote a reply but it seems to have disappeared...)

In general it's better because it checks the validity of the association and not just the presence of the key, but in this case it's equivalent, because the id will always point to an existing achievement with a valid id; no achievement is being created.

Also, `_changed?` methods are only provided for actual attributes, so there is no `achievement_changed?` by default (in contrast, you can do stuff like `validates :achievement, presence: true` on the association).